Example code in lib.rs does not compile

https://github.com/nrc/proc-macro-rules/blob/9562f24a29c7e1807f63e93fa151e9da00fccda2/src/lib.rs#L8-L22

With all the right helper functions in scope:

```console
error[E0382]: use of moved value: `finish`
  --> src/lib.rs:13:30
   |
13 |                 do_something(finish, f, inner, rest[0]);
   |                              ^^^^^^ value moved here, in previous iteration of loop
   |
   = note: move occurs because `finish` has type `proc_macro2::Ident`, which does not implement the `Copy` trait

error[E0382]: use of moved value: `inner`
  --> src/lib.rs:13:41
   |
13 |                 do_something(finish, f, inner, rest[0]);
   |                                         ^^^^^ value moved here, in previous iteration of loop
   |
   = note: move occurs because `inner` has type `std::vec::Vec<proc_macro2::TokenTree>`, which does not implement the `Copy` trait

error[E0507]: cannot move out of borrowed content
  --> src/lib.rs:13:48
   |
13 |                 do_something(finish, f, inner, rest[0]);
   |                                                ^^^^^^^ cannot move out of borrowed content
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -13,6 +13,9 @@ proc-macro-rules-macros = "0.1.1"
 proc-macro2 = "0.4.21"
 syn = { version = "0.15.0", features = ["full", "extra-traits"] }
 
+[dev-dependencies]
+quote = "0.6"
+
 [workspace]
 members = [
     "macros",

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -4,22 +4,50 @@
 //! 
 //! Example:
 //! 
-//! ```rust,ignore
-//! use proc_macro_rules::rules;
+//! ```rust
+//! #![feature(label_break_value, proc_macro_hygiene)]
 //!
-//! rules!(tokens => {
-//!     ($finish:ident ($($found:ident)*) # [ $($inner:tt)* ] $($rest:tt)*) => {
-//!         for f in found {
-//!             do_something(finish, f, inner, rest[0]);
+//! use proc_macro_rules::rules;
+//! use proc_macro2::TokenStream;
+//!
+//! fn main() {
+//!     # use quote::quote;
+//!     # test_rules(quote! { A (B C) #[inner...] ... });
+//!     # test_rules(quote! { foo 1 + 1 });
+//!     # test_rules(quote! { foo });
+//! # }
+//! #
+//! # fn test_rules(tokens: TokenStream) {
+//!     # const IGNORE: &str = stringify! {
+//!     let tokens: TokenStream = /* ... */;
+//!     # };
+//!
+//!     rules!(tokens => {
+//!         ($finish:ident ($($found:ident)*) # [ $($inner:tt)* ] $($rest:tt)*) => {
+//!             for f in found {
+//!                 do_something(&finish, f, &inner, &rest[0]);
+//!             }
 //!         }
-//!     }
-//!     (foo $($bar:expr)?) => {
-//!         match bar {
-//!             Some(e) => foo_with_expr(e),
-//!             None => foo_no_expr(),
+//!         (foo $($bar:expr)?) => {
+//!             match bar {
+//!                 Some(e) => foo_with_expr(e),
+//!                 None => foo_no_expr(),
+//!             }
 //!         }
-//!     }
-//! });
+//!     });
+//! }
+//! #
+//! # use syn::{Expr, Ident};
+//! # use proc_macro2::TokenTree;
+//! #
+//! # fn do_something(
+//! #     finish: &Ident,
+//! #     f: Ident,
+//! #     inner: &[TokenTree],
+//! #     rest: &TokenTree,
+//! # ) {}
+//! # fn foo_with_expr(e: Expr) {}
+//! # fn foo_no_expr() {}
 //! ```
 //! 
 //! Import the `rules` macro with `use proc_macro_rules::rules`, then use with 

